### ISSUE TITLE
false positive bad-reversed-sequence when used with subclass of dict

### ISSUE THREAD

**Issue by davlse** [PyCQA/pylint#4981]
### Bug description

```python
"""reversed(MyDict()) emits bad-reversed-sequence when it should not for Python 3.8+."""
class MyDict(dict):
    """MyDict class"""
reversed(MyDict())
```

### Command used

```shell
pylint mydict.py
```

### Pylint output

```shell
************* Module mydict
mydict.py:5:0: E0111: The first reversed() argument is not a sequence (bad-reversed-sequence)
```

### Expected behavior

No error emitted.

I believe this is an extension of this #3940 issue.

### Pylint version

```shell
pylint 2.10.2
astroid 2.7.3
Python 3.8.5 (default, Oct  8 2020, 09:50:52) 
[GCC 4.8.2]
```
